lib/spack/spack/new_installer.py

```diff
@@ -297,10 +299,16 @@ def worker_function(
     if os.path.exists(spec.prefix):
         shutil.rmtree(spec.prefix)
 
-    # First try to install from binary cache.
-    if mirrors and install_from_buildcache(mirrors, spec, unsigned, state_stream):
-        spack.hooks.post_install(spec, explicit)
-        return
+    if install_policy != "source_only":
```

Comment:
(Nitpick) Add a comment explaining the block, similar to the one liners above
```suggestion
    # Try to install from buildcache, unless user asked for source only
    if install_policy != "source_only":
```